### ISSUE TITLE
sst: generate types without needing to be authenticated

### ISSUE THREAD

**Issue by nguyendon** [anomalyco/sst#5189]
We want to run our typechecker in CI but it needs the autogenerated types in `.sst`. We can only generate these when running `sst dev` or `sst deploy` which requires being authenticated with our provider (AWS). It would be nice to be able to generate the types without needing that.

**Comment by anaydot** [comment]
+1. We ran into something similar and ended up committing our SST types for now.

**Comment by joegearlesss** [comment]
Bump it

**Comment by alexanderbh** [comment]
It would be nice. We have a monorepo setup where sst generates 50+ type files which cluttered up our PRs. So we have only committed the root sst type file and then a script runs in the CI environment which copies this root sst type file into all folders that have a package.json file.

This way we have the root type file (once) for review and it will be copied for typechecking by the CI. Locally sst just generates all the type files but they are gitignored (except the root one)

**Comment by jayair** [comment]
I don't think we can generate the types without building your app. I'll need to double check.

**Comment by apetta** [comment]
Facing a similar issue with CI linting process. It would be nice if the CLI had a type generation command which we could run before running any type-checking.

**Comment by thdxr** [comment]
unfortunately because of the way the underlying system works we cannot support this - there is no guarantee a resource doesn't depend on some derived cloud value before it can even exist
